igor-web/src/main/java/com/netflix/spinnaker/igor/codebuild/AwsCodeBuildAccount.java

```diff
@@ -78,4 +87,28 @@ private String getRoleArn(String accountId, String assumeRole) {
     }
     return assumeRoleValue;
   }
+
+  private List<Artifact> extractArtifactsFromBuild(Build build) {
+    ArrayList<Artifact> artifactsList = new ArrayList<>();
+    BuildArtifacts primaryArtifacts = build.getArtifacts();
+    List<BuildArtifacts> secondaryArtifacts = build.getSecondaryArtifacts();
+    if (primaryArtifacts != null) {
+      artifactsList.add(getS3Artifact(primaryArtifacts.getLocation()));
+    }
+    if (secondaryArtifacts != null) {
+      secondaryArtifacts.forEach(
+          artifacts -> artifactsList.add(getS3Artifact(artifacts.getLocation())));
+    }
+    return artifactsList;
+  }
+
+  private Artifact getS3Artifact(String s3Arn) {
+    String reference = getS3ArtifactReference(s3Arn);
+    return Artifact.builder().type("s3/object").reference(reference).name(reference).build();
```

Comment:
I could be wrong, but I think typically the artifact name is fairly static so that later stages in the pipeline can refer to that name in their definitions, and the version field is used to distinguish between artifacts from different builds.  Can the CodeBuild's namespacing behavior be used here to have a static artifact name and changing version field?

Reply:
Based on the [documentation](https://www.spinnaker.io/reference/artifacts-with-artifactsrewrite/types/s3-object/) `name` should be the full path to the artifact file, beginning with s3://, which is the same as `reference`. `version` is not a valid field for this type of object.

Reply:
Later stages could refer to this artifact by its **display name** defined in UI, which is different from the name here.